### ISSUE TITLE
Fix CI

### ISSUE THREAD

**Issue by cooljeanius** [cooljeanius/gvpe-2.25#2]
Looks like there's some sort of OpenSSL incompatibility...

**Comment by cooljeanius** [comment]
At least, it looks like that when it actually runs...

**Comment by cooljeanius** [comment]
OK I think I was getting ones confused; this one currently looks like an incompatibility with getopt, not openssl:
```
g++ -DHAVE_CONFIG_H -I. -I..  -I.. -I../lib -I../libev  -Wno-unused  -DPKGLIBDIR=/usr/local/lib/gvpe -DCONFDIR=\"/usr/local/etc\" -DLOCALEDIR=\"/usr/local/share/locale\" -DLOCALSTATEDIR=\"/usr/local/var\" -DIFTYPE_native=1 -DIF_linux=1 -Wuninitialized -Wno-unused -std=gnu++98 -g -O2 -MT gvpectrl.o -MD -MP -MF .deps/gvpectrl.Tpo -c -o gvpectrl.o gvpectrl.C
In file included from /usr/include/x86_64-linux-gnu/bits/getopt_posix.h:27,
                 from /usr/include/unistd.h:869,
                 from gvpectrl.C:60:
/usr/include/x86_64-linux-gnu/bits/getopt_core.h:91:12: error: declaration of ‘int getopt(int, char* const*, const char*) throw ()’ has a different exception specifier
   91 | extern int getopt (int ___argc, char *const *___argv, const char *__shortopts)
      |            ^~~~~~
In file included from gvpectrl.C:55:
../lib/getopt.h:108:12: note: from previous declaration ‘int getopt(int, char* const*, const char*)’
  108 | extern int getopt (int argc, char *const *argv, const char *shortopts);
      |            ^~~~~~
gvpectrl.C: In function ‘int keygen(int)’:
gvpectrl.C:293:16: error: aggregate ‘BN_GENCB cb’ has incomplete type and cannot be defined
  293 |       BN_GENCB cb;
      |                ^~
gvpectrl.C:255:12: warning: ignoring return value of ‘int asprintf(char**, const char*, ...)’, declared with attribute warn_unused_result [-Wunused-result]
  255 |   asprintf (&fname, "%s/hostkeys", confbase);
      |   ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gvpectrl.C:259:12: warning: ignoring return value of ‘int asprintf(char**, const char*, ...)’, declared with attribute warn_unused_result [-Wunused-result]
  259 |   asprintf (&fname, "%s/pubkey", confbase);
      |   ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gvpectrl.C:267:16: warning: ignoring return value of ‘int asprintf(char**, const char*, ...)’, declared with attribute warn_unused_result [-Wunused-result]
  267 |       asprintf (&fname, "%s/pubkey/%s", confbase, node->nodename);
      |       ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gvpectrl.C:304:16: warning: ignoring return value of ‘int asprintf(char**, const char*, ...)’, declared with attribute warn_unused_result [-Wunused-result]
  304 |       asprintf (&fname, "%s/hostkeys/%s", confbase, node->nodename);
      |       ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[3]: *** [Makefile:597: gvpectrl.o] Error 1
make[3]: Leaving directory '/home/runner/work/gvpe-2.25/gvpe-2.25/src'
make[2]: *** [Makefile:617: all-recursive] Error 1
make[2]: Leaving directory '/home/runner/work/gvpe-2.25/gvpe-2.25/src'
make[1]: Leaving directory '/home/runner/work/gvpe-2.25/gvpe-2.25'
make[1]: *** [Makefile:437: all-recursive] Error 1
make: *** [Makefile:378: all] Error 2
Error: Process completed with exit code 2.
```

**Comment by cooljeanius** [comment]
fixed the getopt one, so now it's back to the openssl ones again. This document seems relevant: 
https://www.openssl.org/docs/man3.0/man7/migration_guide.html

**Comment by cooljeanius** [comment]
OK, seems to be fixed now as of [6fbec64](https://github.com/cooljeanius/gvpe-2.25/commit/6fbec64182b1294d18e82de23703e3780a8c30d6).
